data-exporter/src/main/java/org/reactome/release/dataexport/UniProtReactomeEntry.java

```diff
@@ -303,6 +373,37 @@ private void setDisplayName(String displayName) {
 		this.displayName = displayName;
 	}
 
+	/**
+	 * Checks if a passed String is a valid canonical UniProt accession (i.e. 6 or 10 alphanumeric characters matching
+	 * the regular expression "[OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9]([A-Z][A-Z0-9]{2}[0-9]){1,2}" as defined at
+	 * https://www.uniprot.org/help/accession_numbers as of October 2019).
+	 * E.g. P12345
+	 * @param accession UniProt accession to test for validity
+	 * @return true if the accession passed is a valid canonical (i.e. not an isoform) UniProt accession,
+	 * false otherwise
+	 */
+	private boolean isValidCanonicalAccession(String accession) {
+		final Pattern ANCHORED_PARENT_ACCESSION_REGEX = Pattern.compile("^" + PARENT_ACCESSION_REGEX.pattern() + "$");
+
+		return ANCHORED_PARENT_ACCESSION_REGEX.matcher(accession).matches();
+	}
+
+	/**
+	 * Checks if a passed String is a valid isoform UniProt accession (i.e. 6 or 10 alphanumeric characters matching
+	 * the regular expression "[OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9]([A-Z][A-Z0-9]{2}[0-9]){1,2}" as defined at
+	 * https://www.uniprot.org/help/accession_numbers as of October 2019, and then followed by a dash and by any number
+	 * of digits).  E.g. P12345-1
+	 * @param accession UniProt accession to test for validity
+	 * @return true if the accession passed is a valid UniProt accession for an isoform, false otherwise
+	 */
+	private boolean isValidIsoformAccession(String accession) {
+		Pattern isoformSuffixRegex = Pattern.compile("-[0-9]+");
```

Comment:
There's a number of `Pattern`s that are used in this class that do not vary. Creating a new pattern that is always the same every time a function is called is not very efficient. If these patterns do not vary then they could be created as `static final` members of the enclosing class.

Reply:
I'll move them to `static final` members.